README.md

```diff
@@ -1,3 +1,301 @@
+* **Build extendable** — System should be extendable for new formats or transformations.
+
+
+## Installation
+
+```sh
+# via npm
+npmi i -DE @yandex/themekit
+# or yarn
+yarn add --dev @yandex/themekit
+```
+
+## Getting start
+
+You can use themekit as cli or node api.
+
+### cli
+
+Build tokens for all formats:
+
+```sh
+themekit build
+```
+
+Call this in the root directory of your project. The only thing needed is a `themekit.config.js` file. There are also arguments:
+
+| Flag               | Description |
+|--------------------|-------------|
+| --config -c [path] | —           |
+| --help             | —           |
+| --version          | —           |
+
+### node
+
+Coming soon.
+
+## Configuration
+
+```ts
+type Config = {
+  src: string
+  platforms?: 'common' | 'deskpad' | 'desktop' | 'touch' | 'touch-pad' | 'touch-phone'
+  formats: {
+    [key: string]: {
+      outDir: string
+      fileName?: string
+      transforms: string[]
+    }
+  }
+}
+```
+
+You can add your own formats, platforms and transforms, see more at [extending](#extending) section.
+
+### platforms
+
+Each platform contains one or more levels, and tokens declared on these levels will be combined in the specified order.
+
+Default set of available platforms:
+
+| platform    | levels                            |
+|-------------|-----------------------------------|
+| common      | common                            |
+| deskpad     | common, deskpad                   |
+| desktop     | common, deskpad, desktop          |
+| touch       | common, touch                     |
+| touch-pad   | common, deskpad, touch, touch-pad |
+| touch-phone | common, touch, touch-phone        |
+
+### formats
+
+Formats are responsible for what files will be created in the end and what they will contain.
+
+#### css.whitepaper
+
+Coming soon.
+
+#### json.whitepaper
+
+Coming soon.
+
+#### css.flat
+
+Coming soon.
+
+#### js.esm
+
+Coming soon.
+
+### transforms
+
+Transforms are responsible for changing the name or value of tokens depending on the type.
+
+#### name.param
+
+Convert token name to `param-case` for usage at `css` or `json`, example:
+
+```sh
+# input
+button_space_all: '20px'
+# output
+button-space-all: '20px'
+```
+
+#### name.const
+
+Convert token name to `CONST_CASE` for usage at `js`, example:
+
+```sh
+# input
+button_space_all: '20px'
+# output
+BUTTON_SPACE_ALL: '20px'
+```
+
+#### color.hex
+
+Convert color to hex if possible, otherwise use rgba scheme, also can process color by hsl scheme, example:
+
+```sh
+# input
+button_bg_color: 'color(#04b h(+22) s(-80%) l(+13%))'
+# output
+button_bg_color: 'rgb(102, 102, 153)'
+```
+
+## Example
+
+More usage cases you can find in [examples][examples], in current example, we will consider of a `css.whitepaper` format.
+
+### prepare fs
+
+The location of the files on the file system is not strictly regulated, at this example is one of the possible options:
```

Comment:
It doesn't matter where you put your files on the fs. Here is our example of how we do it: